src/operator/src/statement/copy_table_to.rs

```diff
@@ -135,24 +137,40 @@ impl StatementExecutor {
             .execute(plan, query_ctx)
             .await
             .context(ExecLogicalPlanSnafu)?;
+
+        let CopyTableRequest {
+            location,
+            connection,
+            ..
+        } = &req;
+        self.copy_to_file(&format, output, location, connection, |path| {
+            debug!("Copy table: {table_id} to path: {path}")
+        })
+        .await
+    }
+
+    pub(crate) async fn copy_to_file<F: FnOnce(String)>(
+        &self,
+        format: &Format,
+        output: Output,
+        location: &str,
+        connection: &HashMap<String, String>,
+        fn_debug: F,
+    ) -> Result<usize> {
         let stream = match output.data {
             OutputData::Stream(stream) => stream,
             OutputData::RecordBatches(record_batches) => record_batches.as_stream(),
             _ => unreachable!(),
         };
 
-        let (_schema, _host, path) = parse_url(&req.location).context(error::ParseUrlSnafu)?;
+        let (_schema, _host, path) = parse_url(location).context(error::ParseUrlSnafu)?;
         let (_, filename) = find_dir_and_filename(&path);
         let filename = filename.context(error::UnexpectedSnafu {
             violated: format!("Expected filename, path: {path}"),
         })?;
-        let object_store =
-            build_backend(&req.location, &req.connection).context(error::BuildBackendSnafu)?;
-        debug!("Copy table: {table_id} to path: {path}");
-        let rows_copied = self
-            .stream_to_file(stream, &format, object_store, &filename)
-            .await?;
-
-        Ok(rows_copied)
+        let object_store = build_backend(location, connection).context(error::BuildBackendSnafu)?;
+        fn_debug(path);
```

Comment:
why need this?

Reply:
supports debugging different information at the original location on the original code

Reply:
I prefer the original styling